sdks/python/apache_beam/testing/util.py

```diff
@@ -112,8 +112,8 @@ def equal_to(expected):
   expected = list(expected)
 
   def _equal(actual):
-    sorted_expected = sorted(expected)
-    sorted_actual = sorted(actual)
+    sorted_expected = sorted(expected, key=str)
```

Comment:
I wonder if we are treating a symptom instead of a rootcause. Do I understand right that the test fails because integers and strings happen to be inside the same collection, which is being sorted?
If that's the case, and if that's a change in behavior from what happens on Python 2, do we understand why the change is happening?

Reply:
Yes the test fails because integers and strings happen to be inside the same collection, which is being sorted. In python 2 sorted will sort different types such as strings and int together while in python 3 this gives a TypeError.
Using the `key=str` as an argument to `sorted` does not solve the problem however because some input fails when using the `str` function. I am not sure what would be a good solution to this?

Reply:
Simply setting key=str is the wrong thing to do here. The contract is that using assert_that(..., equal_to(...)) can only be used on fully orderable PCollections. Likely collections with heterogeneous types are in error, but if we do want to support this we must change the algorithm here. 

One hack would be to partition by type and then sort each bucket. A cheap way to (almost) do this is letting `key=lambda x: (hash(type(x)), type(x), x)`.

Reply:
I updated the code to include key=lambda x: (hash(type(x)), type(x), x,)